packages/devtools_app/test/ansi_up_test.dart

```diff
@@ -88,4 +100,197 @@ void main() {
               '<span style=\'background-color: rgb(168,168,168);color: rgb(255,255,255)\'> 248 </span><span style=\'color: rgb(168,168,168)\'> 248 </span><span style=\'background-color: rgb(178,178,178);color: rgb(255,255,255)\'> 249 </span><span style=\'color: rgb(178,178,178)\'> 249 </span><span style=\'background-color: rgb(188,188,188);color: rgb(255,255,255)\'> 250 </span><span style=\'color: rgb(188,188,188)\'> 250 </span><span style=\'background-color: rgb(198,198,198);color: rgb(255,255,255)\'> 251 </span><span style=\'color: rgb(198,198,198)\'> 251 </span><span style=\'background-color: rgb(208,208,208);color: rgb(255,255,255)\'> 252 </span><span style=\'color: rgb(208,208,208)\'> 252 </span><span style=\'background-color: rgb(218,218,218);color: rgb(255,255,255)\'> 253 </span><span style=\'color: rgb(218,218,218)\'> 253 </span><span style=\'background-color: rgb(228,228,228);color: rgb(255,255,255)\'> 254 </span><span style=\'color: rgb(228,228,228)\'> 254 </span><span style=\'background-color: rgb(238,238,238);color: rgb(255,255,255)\'> 255 </span><span style=\'color: rgb(238,238,238)\'> 255 </span>'));
     });
   });
+
+  group('Logging Screen', () {
+    MockLoggingController mockLoggingController;
+    FakeServiceManager fakeServiceManager;
+    const windowSize = Size(1000.0, 1000.0);
+
+    const totalLogs = 10;
+
+    const nonJsonOutput = 'Non-json details for log number 8';
+    const jsonOutput = '{\n"Details": "of log event 9",\n"logEvent": "9"\n}\n';
+
+    String _ansiCodesOutput() {
+      final sb = StringBuffer();
+      sb.write('Ansi color codes processed for ');
+      final pen = AnsiPen()..rgb(r: 0.8, g: 0.3, b: 0.4, bg: true);
+      sb.write(pen('log 5'));
+      return sb.toString();
+    }
+
+    LogData _generate(int i) {
+      String details = 'log event $i';
+      String kind = 'kind $i';
+      String computedDetails;
+      switch (i) {
+        case 9:
+          computedDetails = jsonOutput;
+          break;
+        case 8:
+          computedDetails = nonJsonOutput;
+          break;
+        case 7:
+          details = null;
+          break;
+        case 5:
+          kind = 'stdout';
+          details = _ansiCodesOutput();
+          break;
+        default:
+          break;
+      }
+
+      final detailsComputer = computedDetails == null
+          ? null
+          : () =>
+              Future.delayed(const Duration(seconds: 1), () => computedDetails);
+      return LogData(kind, details, i, detailsComputer: detailsComputer);
+    }
+
+    final fakeLogData = List<LogData>.generate(totalLogs, _generate);
+
+    Future<void> pumpLoggingScreen(WidgetTester tester) async {
+      await tester.pumpWidget(wrapWithControllers(
+        const LoggingScreenBody(),
+        logging: mockLoggingController,
+      ));
+    }
+
+    setUp(() async {
```

Comment:
I expect a lot of these set up steps are not necessary for the single test that is in this group. Let's remove any that aren't needed. Same comment for the Debugger screen group below.

Reply:
For now I do not have a way to run the tests and verify they pass. Both locally or with bots.
Added TODO for this.